### PR TITLE
 [FIX] p_t_s_biz_ext: Send wrong scale command 

### PR DESCRIPTION
**Pull request by remytms**
Fix the case when modifying a product, the wrong order is send to the
scale system.

For the moment it's not optimised. But it works well.

For reviewing please see only this commit https://github.com/houssine78/addons/commit/42901605ce5b792421762d77bdebaa478c2c9884

### CODE REVIEW

File: product_to_scale_bizerba_extended/models/product.py [hunk 1]
```diff
@@ -118,14 +119,13 @@ def write(self, vals):
...
                 # ticking and unticking the "can be sold" checkbox
                 # trigger the corresponding product_scale_log
-                if 'sale_ok' in vals.keys() or 'active' in vals.keys():
-                    if ((product.sale_ok and product.active)
-                        and (not vals.get('sale_ok', False)
-                             or not vals.get('active', False))):
-                        product._send_to_scale_bizerba('unlink')
-                    elif ((product.sale_ok and vals.get('active', False))
-                          or (product.active and vals.get('sale_ok', False))):
+                if vals.get('active', product.active):
```
Comment (robinkeunen) [inline product_to_scale_bizerba_extended/models/product.py:122]:
```python
if 'active' in vals or product.active:
    ...
```
This is a more pythonesque and human readable way of writing this condition.

File: product_to_scale_bizerba_extended/models/product.py [hunk 2]
```diff
@@ -1,130 +1,143 @@
...
+        send_to_scale = vals.get('scale_group_id', False)
+        res = super(ProductTemplate, self).create(vals)
+        if send_to_scale:
+            product = self.browse(res)
+            self._send_to_scale_bizerba('create')
+        return res
+
+    @api.multi
+    def write(self, vals):
+        defered = {}
```
Comment (robinkeunen) [inline product_to_scale_bizerba_extended/models/product.py:100]: While you're cleaning up, the good spelling is `deferred`.

**Review by robinkeunen — APPROVED** [review]
Looks good to me (see comments) but I don't know bizerba enough to  judge.

File: product_to_scale_bizerba_extended/models/product.py (hunk 1 above)
Comment (houssine78) [reply product_to_scale_bizerba_extended/models/product.py:122]: you expression is not the same. I'll propose to keep the if expression like it is and keep it self explained

**Review by houssine78 — APPROVED** [review]

File: product_to_scale_bizerba_extended/models/product.py (hunk 2 above)
Comment (remytms) [reply product_to_scale_bizerba_extended/models/product.py:100]: Thanks

**Pull request merged**